pydantic/networks.py

```diff
@@ -228,130 +388,123 @@ class MyModel(BaseModel):
-    db: PostgresDsn
-
-    @field_validator('db')
-    def check_db_name(cls, v):
-        assert v.path and len(v.path) > 1, 'database must be provided'
-        return v
-
-m = MyDatabaseModel(db='postgres://user:pass@localhost:5432/foobar')
-print(m.db)
-#> postgres://user:pass@localhost:5432/foobar
-
-try:
-    MyDatabaseModel(db='postgres://user:pass@localhost:5432')
-except ValidationError as e:
-    print(e)
-    '''
-    1 validation error for MyDatabaseModel
-    db
-      Assertion failed, database must be provided
-    assert (None)
-     +  where None = MultiHostUrl('postgres://user:pass@localhost:5432').path [type=assertion_error, input_value='postgres://user:pass@localhost:5432', input_type=str]
-    '''
-```
-"""
+    )
+
+    @property
+    def host(self) -> str:
+        """The required URL host."""
+        ...
+
 
-CockroachDsn = Annotated[
-    Url,
-    UrlConstraints(
+class CockroachDsn(_BaseUrl):
+    """A type that will accept any Cockroach DSN.
+
+    * User info required
+    * TLD not required
+    * Host required
+    """
+
+    _constraints = UrlConstraints(
         host_required=True,
         allowed_schemes=[
             'cockroachdb',
             'cockroachdb+psycopg2',
             'cockroachdb+asyncpg',
         ],
-    ),
-]
-"""A type that will accept any Cockroach DSN.
+    )
 
-* User info required
-* TLD not required
-* Host required
-"""
-AmqpDsn = Annotated[Url, UrlConstraints(allowed_schemes=['amqp', 'amqps'])]
-"""A type that will accept any AMQP DSN.
+    @property
+    def host(self) -> str:
+        """The required URL host."""
+        ...
 
-* User info required
-* TLD not required
-* Host required
-"""
-RedisDsn = Annotated[
-    Url,
-    UrlConstraints(allowed_schemes=['redis', 'rediss'], default_host='localhost', default_port=6379, default_path='/0'),
-]
-"""A type that will accept any Redis DSN.
 
-* User info required
-* TLD not required
-* Host required (e.g., `rediss://:pass@localhost`)
-"""
-MongoDsn = Annotated[MultiHostUrl, UrlConstraints(allowed_schemes=['mongodb', 'mongodb+srv'], default_port=27017)]
-"""A type that will accept any MongoDB DSN.
+class AmqpDsn(_BaseUrl):
+    """A type that will accept any AMQP DSN.
 
-* User info not required
-* Database name not required
-* Port not required
-* User info may be passed without user part (e.g., `mongodb://mongodb0.example.com:27017`).
-"""
-KafkaDsn = Annotated[Url, UrlConstraints(allowed_schemes=['kafka'], default_host='localhost', default_port=9092)]
-"""A type that will accept any Kafka DSN.
+    * User info required
+    * TLD not required
+    * Host not required
+    """
 
-* User info required
-* TLD not required
-* Host required
-"""
-NatsDsn = Annotated[
-    MultiHostUrl,
-    UrlConstraints(allowed_schemes=['nats', 'tls', 'ws', 'wss'], default_host='localhost', default_port=4222),
-]
-"""A type that will accept any NATS DSN.
+    _constraints = UrlConstraints(allowed_schemes=['amqp', 'amqps'])
```

Comment:
https://www.rabbitmq.com/docs/uri-spec

The `host` is "required" but can be zero in length... we previously didn't enforce `host_required` here, so leaving as is, but adding `not` to the docs.